src/core/string_set.cc

```diff
@@ -129,6 +115,29 @@ uint32_t StringSet::ObjExpireTime(const void* str) const {
   return absl::little_endian::Load32(ttlptr);
 }
 
+void StringSet::ObjUpdateExpireTime(const void* obj, uint32_t ttl_sec) {
+  DCHECK_GT(ttl_sec, 0u);  // ttl_sec == 0 would mean find and delete immediately
+  sds str = (sds)obj;
+  char* valptr = str + sdslen(str) + 1;
+
+  uint32_t at = time_now() + ttl_sec;
+  absl::little_endian::Store32(valptr, at);
+}
+
+void StringSet::ObjReplace(const void* obj, uint32_t ttl_sec) {
+  sds str = (sds)obj;
+  sds newsds = MakeSdsWithTtl(str, ttl_sec);
+  bool has_ttl = ttl_sec == UINT32_MAX ? false : true;
+
+  sds prev_entry = (sds)ReplaceObj(newsds, has_ttl);
+  if (prev_entry) {
+    sdsfree(prev_entry);
+  } else {
+    // No previous entry found, so we need to free the new entry.
+    sdsfree(newsds);
+  }
```

Comment:
and we can call `DenseSet::AddOrReplace()` and basically the function behaves like `StringMap::AddOrUpdate()`